django/contrib/postgres/search.py

```diff
@@ -92,14 +92,42 @@ def __init__(self, lhs, connector, rhs, config, output_field=None):
         super(CombinedSearchVector, self).__init__(lhs, connector, rhs, output_field)
 
 
-class SearchQuery(Value):
+class SearchQueryCombinable(object):
+    BITAND = '&&'
+    BITOR = '||'
+
+    def _combine(self, other, connector, reversed, node=None):
+        if not isinstance(other, SearchQueryCombinable):
+            raise TypeError(
+                'SearchQuery can only be combined with other SearchQuerys, got {}'.format(type(other))
+            )
+        if not self.config == other.config:
```

Comment:
I know this patch is merged, but I recently ran into an issue where I tried to combine to `SearchQueries` with different configs and got stopped by this check.

e.g. `SearchQuery('work:*', search_type='raw', config='simple') | SearchQuery('working', config='english')`

In SQL it seems to be possible, (e.g. `SELECT to_tsquery('simple', 'work:*') || plainto_tsquery('english', 'working') > 'work:*' | 'work'`).

So I was wondering, is there any reason to enforce this check? Or even set `self.config ` on `CombinedSearchQuery` at all?

Reply:
https://github.com/django/django/pull/10704 proposes to remove this.